lib/main_common.pm

```diff
@@ -568,47 +570,69 @@ sub load_system_role_tests {
         loadtest "installation/installer_desktopselection";
     }
 }
-sub load_jeos_tests {
-    if (get_var('JEOS_OPENSTACK')) {
-        loadtest 'boot/boot_to_desktop';
-        if (get_var('JEOS_OPENSTACK_UPLOAD_IMG')) {
-            loadtest "publiccloud/upload_image";
-        }
-        elsif (get_var('JEOS_OPENSTACK_CHECK_BOOT')) {
-            loadtest "jeos/openstack_check_boot";
-        }
+
+sub load_jeos_openstack_tests {
+    return unless is_openstack;
+    my $args = OpenQA::Test::RunArgs->new();
+
+    loadtest 'boot/boot_to_desktop';
+    if (get_var('JEOS_OPENSTACK_UPLOAD_IMG')) {
+        loadtest "publiccloud/upload_image";
+        return;
     } else {
-        if ((is_arm || is_aarch64) && is_opensuse()) {
-            # Enable jeos-firstboot, due to boo#1020019
-            load_boot_tests();
-            loadtest "jeos/prepare_firstboot";
-        }
-        load_boot_tests();
-        loadtest "jeos/firstrun";
-        loadtest "jeos/record_machine_id";
-        loadtest "console/system_prepare" if is_sle;
-        loadtest "console/force_scheduled_tasks";
-        unless (get_var('INSTALL_LTP')) {
-            loadtest "jeos/grub2_gfxmode";
-            loadtest "jeos/diskusage";
-            loadtest "jeos/build_key";
-            loadtest "console/prjconf_excluded_rpms";
-        }
-        unless (get_var('CONTAINER_RUNTIME')) {
-            loadtest "console/journal_check";
-            loadtest "microos/libzypp_config";
-        }
-        if (is_sle) {
-            loadtest "console/suseconnect_scc";
-            loadtest "jeos/efi_tid" if (get_var('UEFI') && is_sle('=12-sp5'));
-        }
+        loadtest "jeos/prepare_openstack", run_args => $args;
+        loadtest 'publiccloud/ssh_interactive_start', run_args => $args;
+    }
+
+    loadtest "jeos/record_machine_id";
```

Comment:
I wonder if we can reuse somehow `load_jeos_tests` here, as both methods are quite similar at this stage. Of course, we would need to do things like `load_boot_tests unless is_openstack` and similar in other places... not sure if it's worth it or not..

Reply:
Yes, we can however then the function becomes hard to read. I fancy wrapper load function as it is used in https://github.com/os-autoinst/os-autoinst-distri-opensuse/blob/master/lib/main_publiccloud.pm#L35. The thing is to keep it somehow concise and our tests need to end with `ssh_interactive_end` that will be loaded with exactly the same `run_args` object as `ssh_interactive_start'`

Reply:
ok, let's keep it easier to read and not pollute legacy function which is used for openSUSE as well. Thanks!